BinaryTree.java

```diff
@@ -0,0 +1,198 @@
+package com.katermar.assignment2;

```

Comment:
Classes that belong to different subpackages usually live in separate subfolders, so that you can have two classes with the same name, each in diffrent package.

Proper path for BinaryTree.java would be com/katermar/assignment2/BinaryTree.java. Could you please move all .java files to proper subfolders? It is quite [easy](https://help.github.com/articles/moving-a-file-to-a-new-location/).